src/acquire.c

```diff
@@ -25,72 +25,87 @@
 
 void acquire_process(acquire_t *st)
 {
-    float complex max_v = 0;
+    float complex max_v = 0, phase_increment;
     float angle, angle_diff, angle_factor, max_mag = -1.0f;
-    unsigned int samperr = 0, i, j, keep;
+    int samperr = 0;
+    unsigned int i, j, keep;
     unsigned int mink = 0, maxk = FFTCP;
 
     if (st->idx != FFTCP * (M + 1))
         return;
 
     if (st->input->sync.ready)
     {
-        mink = FFTCP / 2 - 25;
-        maxk = FFTCP / 2 + 25;
-    }
+        samperr = FFTCP / 2 + st->input->sync.samperr;
+        st->input->sync.samperr = 0;
 
-    memset(st->sums, 0, sizeof(float complex) * FFTCP);
-    for (i = mink; i < maxk + CP; ++i)
-    {
-        for (j = 0; j < M; ++j)
-            st->sums[i] += st->buffer[i + j * FFTCP] * conjf(st->buffer[i + j * FFTCP + FFT]);
+        angle_diff = -st->input->sync.angle;
+        st->input->sync.angle = 0;
+        angle = st->prev_angle + angle_diff;
+        st->prev_angle = angle;
```

Comment:
The post-FFT timing & frequency error estimates are used once synchronization is achieved, thereby avoiding the need to compute the (less accurate) pre-FFT estimates.